astropy/io/fits/tests/test_table.py

```diff
@@ -1753,6 +1755,19 @@ def test_multi_dimensional_columns(self):
         assert t.field(1).dtype.str[-1] == '5'
         assert t.field(1).shape == (3, 4, 3)
 
+    def test_oned_array_single_element(self):
+        # a table with rows that are 1d arrays of a single value
+        data = np.array([(1, ), (2, )], dtype=([('x', 'i4', (1, ))]))
+        thdu = fits.BinTableHDU.from_columns(data)
+
+        thdu.writeto(self.temp('onedtable.fits'))
+
+        with fits.open(self.temp('onedtable.fits')) as hdul:
+            thdu = hdul[1]
+
+            c = thdu.data.field(0)
+            assert c.shape == (2, 1)
+
```

Comment:
Could you also add a check for `TDIM` ?

Reply:
Also it could be useful to add a test with `astropy.table.Table` (in `test_connect.py`) to make sure the use case of your original issue is fixed.

Reply:
done